includes/addons.php

```diff
@@ -271,28 +274,39 @@ function pmpro_admin_init_updating_plugins() {
 
 	// updating one or more plugins via Dashboard -> Upgrade
 	if ( basename( $_SERVER['SCRIPT_NAME'] ) == 'update.php' && ! empty( $_REQUEST['action'] ) && $_REQUEST['action'] == 'update-selected' && ! empty( $_REQUEST['plugins'] ) ) {
-		// figure out which plugin we are updating
+		// figure out which plugins we are updating
 		$plugins = explode( ',', stripslashes( $_GET['plugins'] ) );
 		$plugins = array_map( 'urldecode', $plugins );
 
 		// look for addons
-		$plus_addons = array();
-		$plus_plugins = array();
+		$premium_addons = array();
+		$premium_plugins = array();
 		foreach ( $plugins as $plugin ) {
 			$slug = str_replace( '.php', '', basename( $plugin ) );
 			$addon = pmpro_getAddonBySlug( $slug );
-			if ( ! empty( $addon ) && $addon['License'] == 'plus' ) {
-				$plus_addons[] = $addon['Name'];
-				$plus_plugins[] = $plugin;
+			if ( ! empty( $addon ) && pmpro_license_type_is_premium( $addon['License'] ) ) {
+				if ( ! isset( $premium_addons[$addon['License']] ) ) {
+					$premium_addons[$addon['License']] = array();
+					$premium_plugins[$addon['License']] = array();
+				}
+				$premium_addons[$addon['License']][] = $addon['Name'];
+				$premium_plugins[$addon['License']][] = $plugin;
 			}
 		}
 		unset( $plugin );
 
 		// if Plus addons found, check license key
-		if ( ! empty( $plus_plugins ) && ! pmpro_license_isValid( null, 'plus' ) ) {
-			// show error
-			$msg = __( 'You must have a <a href="https://www.paidmembershipspro.com/pricing/?utm_source=wp-admin&utm_pluginlink=bulkupdate">valid PMPro Plus License Key</a> to update PMPro Plus add ons. The following plugins will not be updated:', 'paid-memberships-pro' );
-			echo '<div class="error"><p>' . $msg . ' <strong>' . implode( ', ', $plus_addons ) . '</strong></p></div>';
+		if ( ! empty( $premium_plugins ) ) {			
+			foreach( $premium_plugins as $license_type => $premium_plugin ) {				
+				// if they have a good license, skip the error				
+				if ( pmpro_can_download_addon_with_license( $license_type ) ) {
+					continue;
+				}
+				
+				// show error
+				$msg = sprintf( __( 'You must have a <a href="https://www.paidmembershipspro.com/pricing/?utm_source=wp-admin&utm_pluginlink=bulkupdate">valid PMPro %s License Key</a> to update PMPro %s add ons. The following plugins will not be updated:', 'paid-memberships-pro' ), ucwords( $license_type ), ucwords( $license_type ) );
```

Comment:
It's painful but we should reconsider usage of HTML in the translation text like this at some point, see:

https://core.trac.wordpress.org/changeset/45542

It's okay for now but we need to approach our strings with more consideration that we don't want HTML in the output of the strings.

Reply:
Will fix in a separate PR.